mesonbuild/modules/simd.py

```diff
@@ -13,76 +13,110 @@
+                  allow_unknown=True) # Because we also accept STATIC_LIB_KWS, but we check them in the interpreter call later on.
+    def check(self, state: ModuleState, args: T.Tuple[str], kwargs: CheckKw) -> T.List[T.Union[T.List[build.StaticLibrary], build.ConfigurationData]]:
         if 'sources' in kwargs:
             raise mesonlib.MesonException('SIMD module does not support the "sources" keyword')
-        basic_kwargs = {}
-        for key, value in kwargs.items():
-            if key not in self.isets and key != 'compiler':
-                basic_kwargs[key] = value
+
+        result: T.List[build.StaticLibrary] = []
+        prefix = args[0]
         compiler = kwargs['compiler']
-        if not isinstance(compiler, compilers.compilers.Compiler):
-            raise mesonlib.MesonException('Compiler argument must be a compiler object.')
         conf = build.ConfigurationData()
-        for iset in self.isets:
-            if iset not in kwargs:
+
+        local_keys = set((*ISETS, 'compiler'))
+        static_lib_kwargs = {k: v for k, v in kwargs.items() if k not in local_keys}
+
+        for iset in ISETS:
+            sources = kwargs[iset] # type: ignore
+
+            cargs = compiler.get_instruction_set_args(iset)
+            if cargs is None:
+                mlog.log(f'Compiler supports {iset}:', mlog.red('NO'))
                 continue
-            iset_fname = kwargs[iset] # Might also be an array or Files. static_library will validate.
-            args = compiler.get_instruction_set_args(iset)
-            if args is None:
-                mlog.log('Compiler supports %s:' % iset, mlog.red('NO'))
+
+            if not compiler.has_multi_arguments(cargs, state.environment)[0]:
+                mlog.log(f'Compiler supports {iset}:', mlog.red('NO'))
                 continue
-            if args:
-                if not compiler.has_multi_arguments(args, state.environment)[0]:
-                    mlog.log('Compiler supports %s:' % iset, mlog.red('NO'))
-                    continue
-            mlog.log('Compiler supports %s:' % iset, mlog.green('YES'))
-            conf.values['HAVE_' + iset.upper()] = ('1', 'Compiler supports %s.' % iset)
-            libname = prefix + '_' + iset
-            lib_kwargs = {'sources': iset_fname,
-                          }
-            lib_kwargs.update(basic_kwargs)
-            langarg_key = compiler.get_language() + '_args'
-            old_lang_args = mesonlib.extract_as_list(lib_kwargs, langarg_key)
-            all_lang_args = old_lang_args + args
-            lib_kwargs[langarg_key] = all_lang_args
-            result.append(self.interpreter.func_static_lib(None, [libname], lib_kwargs))
+
+            mlog.log(f'Compiler supports {iset}:', mlog.green('YES'))
+            conf.values['HAVE_' + iset.upper()] = ('1', f'Compiler supports {iset}.')
+
+            my_name = f'{prefix}_{iset}'
+
+            my_kwargs = static_lib_kwargs.copy()
```

Comment:
You need to somehow get this to `kwargs.StaticLibrary`. I think you could inherit that instead of `TypedDict` for `CheckKwargs`, or you could cast here.

Reply:
I think that inheriting will make things easier long term